docs/docs/ref-03-component-specs.md

```diff
@@ -148,6 +148,8 @@ componentWillReceiveProps: function(nextProps) {
 
 > Note:
 >
+> It is **incorrect** for code executed during this lifecycle method to assume that props have changed.  To understand why, read [A implies B does not imply B implies A](/react/blog/2016/01/08/A-implies-B-does-not-imply-B-implies-A.html)
```

Comment:
Instead of saying what's incorrect, I would suggest to make the explanation at the top more verbose. Now it just says: "Invoked when a component is receiving new props." That this sentence is formal logic is not obvious.